pkg/shares/compact_shares_test.go

```diff
@@ -112,6 +112,23 @@ func Test_processCompactShares(t *testing.T) {
 	}
 }
 
+func TestParseOutOfContextShares(t *testing.T) {
```

Comment:
[optional] in addition to this test, should we consider adding a test case for https://github.com/rootulp/celestia-app/blob/01282c42a8b1d7ea2cfd325c2ed10aed4c317966/pkg/shares/parse.go#L12 that tests out of context shares?

Motivation: `parseCompactShares` is private and `ParseTxs` is public.

Reply:
Since`ParseTxs` uses `parseCompactShares` under the hood so instead of adding another test, modified this test using `ParseTxs`to make it simpler.